app/src/main/java/com/codesoom/assignment/validate/TodoValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.codesoom.assignment.validate;
+
+import com.codesoom.assignment.controller.TodoHttpController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TodoValidator {
+    final private TodoHttpController todoHttpController;
+    final private List<String> exceptionIdList;
+
+    public TodoValidator() {
+        exceptionIdList = new ArrayList<>();
+        exceptionIdList.add("null");
+        exceptionIdList.add("undefined");
+        todoHttpController = new TodoHttpController();
+    }
+
+    public void isContainsInvalidId(String path) {
+        if (path.split("/").length > 1) {
+            String id = path.split("/")[1];
+            if (id.equals("null") || id.equals("undefined")) {
+                throw new IllegalArgumentException();
+            }
+        }
+    }
```

Comment:
이 메소드 안쪽의 중첩된 if 문을 early return 방식으로 리팩터링 해보세요.

Reply:
@johngrib 
이건 잘 이해가 안되는데...
중첩 if문을 지우라고 하셨으면, 
```java
if (path.split("/").length > 1 && (path.split("/")[1].equals("null") || path.split("/")[1].equals("undefined"))) {
```
라고 변경했을거 같은데, early return방식으로 어찌 처리해야할지 잘 모르겠습니다.
```java
if (!id.equals("null") && !id.equals("undefined")) {
                return;
}
throw new IllegalArgumentException();
```
위와 같은걸 말씀하신건 아닌거같은데... 힌트좀 주실 수 있을까요?

Reply:
if 문 안쪽의 조건을 바꿔서 else 에 해당하는 분기가 먼저 오도록 해주면 됩니다.

예를 들어서 다음과 같이 아무것도 하지 않는 else를 추가할 수 있겠죠?

```suggestion
    public void isContainsInvalidId(String path) {
        if (path.split("/").length > 1) {
            String id = path.split("/")[1];
            if (id.equals("null") || id.equals("undefined")) {
                throw new IllegalArgumentException();
            }
        } else {
            // 아무것도 안함
            return;
        }
    }
```

이제 아무것도 안하는 분기를 위로 올려주면 됩니다.

```suggestion
    public void isContainsInvalidId(String path) {
        if (path.split("/").length <= 1) {
            // 아무것도 안함
            return;
        }

        String id = path.split("/")[1];
        if (id.equals("null") || id.equals("undefined")) {
            throw new IllegalArgumentException();
        }
    }
```

이제 뭔가 바람직하지 않은 값이 들어왔을 때 바로바로 return으로 메소드를 탈출해주는 방식으로 수정이 되었습니다.